app/models/job.rb

```diff
@@ -61,7 +61,7 @@ def duration
   end
 
   def can_be_cancelled_by?(user)
-    started_by?(user) || user.admin? || user.admin_for?(project)
+    user.admin? || user.deployer_for?(project)
```

Comment:
I was trying not to change the behaviour too much from what it currently is, so I thought it made sense for any admin and any deployer of the project to be able to cancel the deploy. But I don't have a clear view of the difference between these roles in Samson. Do you think it would be better for any deployer at all to be able to cancel a deploy?

Reply:
`deployer?` means `global_deployer?` so yes :)